semgrep-core/Core/Caching.ml

```diff
@@ -71,6 +168,22 @@ let diff_backrefs bound_metavars ~new_backref_counts ~old_backref_counts =
    This function decorates the pattern AST with the set of metavariables
    that are used in the rest of the pattern. This set is consulted at runtime
    to determine whether a captured metavariable should go into the cache key.
+
+   We only consult the cache at strategic locations:
+   - on an ellipsis pattern, for every position in the target stmt list,
+   - and only if the rest of the pattern doesn't contain free metavariables.
```

Comment:
Is this supposed to say "bound metavariables"? (Or am I confused about something)

Reply:
I'll clarify.

Reply:
I think my use of the terms "free" and "bound" is just wrong.

Reply:
Anyway, the notion is that `let x = 1 in x + 1` is not parametrized whereas `x + 1` alone is parametrized by `x`.

Reply:
Changing it to: "and only if the rest of the pattern is not parametrized by earlier captures"